code/modules/mob/living/carbon/human/species_types/ethereal.dm

```diff
@@ -73,8 +83,95 @@
 	UnregisterSignal(C, COMSIG_ATOM_EMAG_ACT)
 	UnregisterSignal(C, COMSIG_ATOM_EMP_ACT)
 	QDEL_NULL(ethereal_light)
+	if(rooting)
+		rooting.Remove(C)
 	return ..()
 
+/datum/action/innate/root
+	name = "Root"
+	desc = "Root into good soil to gain charge."
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon_state = "plant-22"
+	icon_icon = 'icons/obj/flora/plants.dmi'
+	background_icon_state = "bg_alien"
+
+/datum/action/innate/root/Activate()
+	var/mob/living/carbon/human/H = owner
+	var/datum/species/elzuose/E = H.dna.species
+	var/turf/terrain = get_turf(H)
+	// this is healthy for elzu, they shouldnt be able to overcharge and get heart attacks from this
+	var/charge_limit = ELZUOSE_CHARGE_FULL
+	var/obj/item/organ/stomach/ethereal/stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
+
+	if(H.wear_suit && istype(H.wear_suit, /obj/item/clothing))
+		var/obj/item/clothing/CS = H.wear_suit
+		if (CS.clothing_flags & THICKMATERIAL)
+			to_chat(H, span_warning("Your [CS.name] is too thick to root in!"))
+			return
+
+	if(stomach.crystal_charge > charge_limit)
+		to_chat(H,span_warning("Your charge is full!"))
+		return
+	E.drain_time = world.time + E.root_time
+	H.visible_message(span_notice("[H] is digging into the ground"),span_warning("You start to dig yourself into the ground to root. You won't won't be able to move once you start the process."),span_notice("You hear digging."))
+	if(!do_after(H,E.dig_time, target = H))
+		to_chat(H,span_warning("You were interupted!"))
+		return
+	ADD_TRAIT(H,TRAIT_IMMOBILIZED,SPECIES_TRAIT)
+	ADD_TRAIT(H,TRAIT_PUSHIMMUNE,SPECIES_TRAIT)
+	to_chat(H, span_notice("You root into the ground and begin to feed."))
+
+	while(do_after(H, E.root_time, target = H))
+		if(istype(stomach))
+			to_chat(H, span_notice("You receive some charge from rooting."))
+			stomach.adjust_charge(E.root_charge_gain)
+			H.adjustBruteLoss(-3)
+			H.adjustFireLoss(-3)
+			SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "root", /datum/mood_event/root)
+
+			if(stomach.crystal_charge > charge_limit)
+				stomach.crystal_charge = ELZUOSE_CHARGE_FULL
+				H.visible_message(span_notice("[H] is digging out of the ground."),span_notice("You finish rooting and begin digging yourself out."),span_notice("You hear digging."))
+				E.digout(H, E)
+		else
+			to_chat(H,span_warning("You can't recieve charge from rooting!"))
+			break
+
+	// Your rooting do afters weren't completed
+	while(HAS_TRAIT_FROM(H, TRAIT_IMMOBILIZED, SPECIES_TRAIT))
```

Comment:
I also realize. I don't think it reaches the while loop at present does it? It's stuck in that do_after of healing?

Reply:
It hits the while loop when the first do_after is interrupted. If you get moved, change items in hand, etc.

Reply:
Alright should of thought of that by my other point still mostly stands. Are they meant to be completely mobile during this? it seems to be they would be able to still interact with things in there hands as they are only rooted by the legs/chest. I see your reason for the while loop is to make sure they don't cancel the digout. If they do cancel the digout they are now stuck in a state where they are still rooted but no longer charging, caused by doing very minor things like switching hands.

Reply:
Originally I did have their hands blocked for the duration of the rooting, but I thought it would be more helpful for them to have a way to end their rooting early by shifting items in hand since they don't have the option of moving. Switching hands alone shouldn't interrupt the do afters AFAIK, but moving items in/out of them should.

A status effect does sound good though, I'll look into that.